contracts/actions/OperationRunner.sol

```diff
@@ -0,0 +1,58 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity >=0.7.6;
+pragma abicoder v2;
+
+import "../interfaces/mcd/IJoin.sol";
+import "../interfaces/mcd/IManager.sol";
+import "./ActionBase.sol";
+import "hardhat/console.sol";
+import "../ServiceRegistry.sol";
+
+struct Operation {
+  string name;
+  bytes[][] callData;
```

Comment:
why it is multidimentional? I'm quite sure we can get away with one dimention and then do abi.decode
similar trick like:
https://github.com/OasisDEX/multiply-proxy-actions/blob/c53cc9d0b272ea997c56d06c8b1fe4b1aac7b5fe/contracts/multiply/GuniMultiplyProxyActions.sol#L264

generally it is reasonable to assume that action knows format of its parameters, so You pass bytes to action and it decodes it in any way it wants